src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -611,7 +612,13 @@ private void updateQueryStats(VectorQueryType vectorQueryType) {
         }
     }
 
-    private float[] getVectorForCreatingQueryRequest(VectorDataType vectorDataType, KNNEngine knnEngine) {
+    private float[] getVectorForCreatingQueryRequest(VectorDataType vectorDataType, KNNEngine knnEngine, SpaceType spaceType) {
+
+        // Cosine similarity is supported as Inner product by FAISS by normalizing input vector, hence, we have to normalize
```

Comment:
Can we move this check out of this class? This class is already very crowded and I want to avoid adding more checks around engines. Instead, could we investigate either adding it to https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/engine/KNNLibrarySearchContext.java and/or adding a method in KNNVectorFieldType (https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java#L85) that says "should normalize query" or, better yet, transformQuery?

Reply:
Moved it to KNNVectorFieldType